### PR TITLE
assembly.c: mono_stringify_assembly_name,  reflection.c: assembly_nam…

### PR DESCRIPTION
**Pull request by alexger97**
Good afternoon, I apologize if something is wrong, I am new to large projects, but faced with a problem: https://github.com/mono/mono/issues/19803
I modified, it seems to me, the code so that the problem disappears

### CODE REVIEW

File: mono/metadata/assembly.c [hunk 1]
```diff
@@ -1251,15 +1251,21 @@ mono_stringify_assembly_name (MonoAssemblyName *aname)
...
+	char *token_string = "";
+
+	if (aname->culture [0])
+		culture_string = g_strdup_printf (", Culture=%s", aname->culture);
+	if (aname->public_key_token [0])
+		token_string = g_strdup_printf (", PublicKeyToken=%s%s", aname->public_key_token,(aname->flags & ASSEMBLYREF_RETARGETABLE_FLAG) ? ", Retargetable=Yes" : "");
+	if (aname->has_version)
+		version_string = g_strdup_printf (", Version=%d.%d.%d.%d", aname->major, aname->minor, aname->build, aname->revision);
+
+	return g_strdup_printf ("%s%s%s%s%s%s", quote, aname->name, quote, version_string, culture_string, token_string);
```
Comment (lambdageek) [inline mono/metadata/assembly.c:1265]: There's a problem here: each call to `g_strdup_printf` allocates memory that needs to be freed.  So one option would be to make sure to call `g_free` on each of `version_string`, `culture_string` and `token_string` before returning.

But a cleaner way is to use the `GString` API:
```c
   GString *str;
   str = g_string_new ();
   g_string_append_printf (str, "%s%s%s", quote, aname->name, quote);
   if (aname->culture[0])
        g_string_append_printf (str, ", Culture=%s", aname->culture);
   ...
   if (aname->has_version)
        g_string_append_printf (", Version=%d.%d.%d.%d", aname->major, ..., aname->revision);

   char *result = g_string_free (str, FALSE); /* result is the final formatted string */
   return result;
```

File: mono/metadata/reflection.c [hunk 2]
```diff
@@ -1501,26 +1501,48 @@ assembly_name_to_aname (MonoAssemblyName *assembly, char *p) {
 	int found_sep;
 	char *s;
 	gboolean quoted = FALSE;
+	int i=strlen(p)-1;
```
Comment (jaykrell) [inline mono/metadata/reflection.c:1504]: Spaces around binary operators, here and elsewhere.

File: mono/metadata/reflection.c [hunk 3]
```diff
@@ -1501,26 +1501,48 @@ assembly_name_to_aname (MonoAssemblyName *assembly, char *p) {
...
 	if (quoted) {
 		if (*p != '"')
 			return 1;
 		*p = 0;
 		p++;
 	}
 	if (*p != ',')
+	{
+	strcpy(s,assembly->name);
+	for (int i=0;i<strlen(s);i++)
```
Comment (jaykrell) [inline mono/metadata/reflection.c:1537]: Take strlen out of loop.

File: mono/metadata/assembly.c (hunk 1 above)
Comment (alexger97) [reply mono/metadata/assembly.c:1265]: Thank you ! But little mistake in g_string_new () -  
need to pass to the constructor   gchar * . In this case, I think the best is NULL

**Pull request closed without merging**